econia-labs/econia
# [ECO-1542] add price levels to mqtt

# Description

This PR adds price levels to MQTT.

<details>
<summary>The <code>$topic $payload</code> looks like this:</summary>

```
levels/1/ask/1 {"price":15,"size":1000000}
levels/1/bid/1 {"price":14,"size":100000}
levels/6/ask/1 {"price":6,"size":10000}
levels/7/ask/1 {"price":7390,"size":2870048}
levels/7/ask/2 {"price":7399,"size":3343432}
levels/7/ask/3 {"price":7400,"size":90551}
levels/7/ask/4 {"price":7414,"size":189771}
levels/7/ask/5 {"price":7418,"size":2006879}
levels/7/ask/6 {"price":7433,"size":100000}
levels/7/ask/7 {"price":7555,"size":2000}
levels/7/ask/8 {"price":7580,"size":60000}
levels/7/ask/9 {"price":8400,"size":10000}
levels/7/ask/10 {"price":20000,"size":2000}
levels/7/bid/1 {"price":7367,"size":2779662}
levels/7/bid/2 {"price":7360,"size":974}
levels/7/bid/3 {"price":7357,"size":243365}
levels/7/bid/4 {"price":7347,"size":1011931}
levels/7/bid/5 {"price":7345,"size":977932}
levels/7/bid/6 {"price":7309,"size":100000}
levels/7/bid/7 {"price":7108,"size":200000}
levels/7/bid/8 {"price":6965,"size":200000}
levels/7/bid/9 {"price":6950,"size":4500}
levels/7/bid/10 {"price":6900,"size":500}
```

</details>

# Testing

To test this PR, you need to:

- Add `MQTT_PRICE_LEVELS="yes"` to your DSS docker `.env` file to enable this feature (disabled by default).
- Start the DSS and go to a remote location where you can enjoy nature's beauty while the DSS syncs a bit.
- Ensure you have `mosquitto` installed to use it's CLI tools.
- Use `mosquitto_sub --verbose -t levels/# -h localhost -p 21883` to check that you are getting price levels events.

# Checklist

- [x] Did you add tests to cover new code or a fixed issue? (Manually tested)
- [x] Did you check off all checkboxes from the linked Linear task? (Ignore if
  you are not a member of Econia Labs)
